From 77eb8e2a40d97d49df24793736a8d9feef485322 Mon Sep 17 00:00:00 2001
From: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date: Tue, 13 Sep 2016 14:47:46 +0200
Subject: [PATCH 1/2] Move old_style_globals to individual modules

---
 src/sage/calculus/var.pyx                   | 2 ++
 src/sage/ext/interactive_constructors_c.pyx | 1 +
 src/sage/misc/citation.pyx                  | 5 ++++-
 src/sage/misc/fpickle.pyx                   | 1 +
 src/sage/misc/reset.pyx                     | 5 ++++-
 src/sage/misc/sage_timeit_class.pyx         | 4 ++--
 src/sage/structure/category_object.pyx      | 1 +
 src/sage/structure/sage_object.pyx          | 1 +
 src/setup.py                                | 4 ----
 9 files changed, 16 insertions(+), 8 deletions(-)

Note: In addition to this sagemath commit, this patch also contains a change to setup.py
regarding the missing Cython.Compiler.Options.directive_defaults not being defined anymore. --TH

--- a/sage/src/sage/calculus/var.pyx
+++ b/sage/src/sage/calculus/var.pyx
@@ -1,6 +1,8 @@
+# cython: old_style_globals=True
 """
 Symbolic variables
 """
+
 from sage.symbolic.function_factory import function as new_function
 from sage.symbolic.ring import SR
 
--- a/sage/src/sage/ext/interactive_constructors_c.pyx
+++ b/sage/src/sage/ext/interactive_constructors_c.pyx
@@ -1,3 +1,4 @@
+# cython: old_style_globals=True
 r"""
 Constructors that automatically inject variables into the global module scope
 """
--- a/sage/src/sage/misc/citation.pyx
+++ b/sage/src/sage/misc/citation.pyx
@@ -1,4 +1,7 @@
-"Dependency usage tracking for citations"
+# cython: old_style_globals=True
+"""
+Dependency usage tracking for citations
+"""
 
 from sage.misc.all import tmp_filename
 from sage.env import SAGE_ROOT
--- a/sage/src/sage/misc/fpickle.pyx
+++ b/sage/src/sage/misc/fpickle.pyx
@@ -1,3 +1,4 @@
+# cython: old_style_globals=True
 """
 Function pickling
 
--- a/sage/src/sage/misc/reset.pyx
+++ b/sage/src/sage/misc/reset.pyx
@@ -1,4 +1,7 @@
-"Interpreter reset"
+# cython: old_style_globals=True
+"""
+Interpreter reset
+"""
 
 import sys
 
--- a/sage/src/sage/misc/sage_timeit_class.pyx
+++ b/sage/src/sage/misc/sage_timeit_class.pyx
@@ -1,3 +1,5 @@
+# This is here in Cython so we can access the interpreter globals
+# cython: old_style_globals=True
 """
 The ``timeit`` command
 
@@ -5,8 +7,6 @@
 """
 from __future__ import absolute_import
 
-# This is here in Cython so we can get the interpreter globals
-
 from . import sage_timeit
 
 
--- a/sage/src/sage/structure/category_object.pyx
+++ b/sage/src/sage/structure/category_object.pyx
@@ -1,3 +1,4 @@
+# cython: old_style_globals=True
 r"""
 Base class for objects of a category
 
--- a/sage/src/sage/structure/sage_object.pyx
+++ b/sage/src/sage/structure/sage_object.pyx
@@ -1,4 +1,5 @@
 # -*- encoding: utf-8 -*-
+# cython: old_style_globals=True
 r"""
 Abstract base class for Sage objects
 """
--- a/sage/src/setup.py
+++ b/sage/src/setup.py
@@ -562,16 +562,7 @@
     import Cython.Compiler.Options
     import Cython.Compiler.Main
 
-    # Sage uses these directives (mostly for historical reasons).
     Cython.Compiler.Options.embed_pos_in_docstring = True
-    Cython.Compiler.Options.directive_defaults['autotestdict'] = False
-    Cython.Compiler.Options.directive_defaults['cdivision'] = True
-    Cython.Compiler.Options.directive_defaults['fast_getattr'] = True
-    # The globals() builtin in Cython was fixed to return to the current scope,
-    # but Sage relies on the broken behavior of returning to the nearest
-    # enclosing Python scope (e.g. to perform variable injection).
-    Cython.Compiler.Options.old_style_globals = True
-
     debug = False
     if os.environ.get('SAGE_DEBUG', None) != 'no':
         print('Enabling Cython debugging support')
@@ -608,7 +599,10 @@
         force=force,
         aliases=aliases,
         compiler_directives={
+            'autotestdict': False,
+            'cdivision': True,
             'embedsignature': True,
+            'fast_getattr': True,
             'profile': profile,
         })
 
